src/WSRequestHandler_General.cpp

```diff
@@ -181,3 +181,84 @@ HandlerResponse WSRequestHandler::HandleGetStats(WSRequestHandler* req) {
 	obs_data_set_obj(response, "stats", stats);
 	return req->SendOKResponse(response);
 }
+
+/**
+ * Get basic OBS video information
+ * 
+ * @return {int} `baseWidth` Base (canvas) width
+ * @return {int} `baseHeight` Base (canvas) height
+ * @return {int} `outputWidth` Output width
+ * @return {int} `outputHeight` Output height
+ * @return {String} `scaleType` Scaling method used if output size differs from base size
+ * @return {double} `fps` Frames rendered per second
+ * @return {String} `videoFormat` Video color format
+ * @return {String} `colorSpace` Color space for YUV
+ * @return {String} `colorRange` Color range (full or partial)
+ * 
+ * @api requests
+ * @name GetVideoInfo
+ * @category general
+ * @since 4.6.0 
+ */
+HandlerResponse WSRequestHandler::HandleGetVideoInfo(WSRequestHandler* req) {
+	obs_video_info ovi;
+	obs_get_video_info(&ovi);
+	OBSDataAutoRelease response = obs_data_create();
+	obs_data_set_int(response, "baseWidth", ovi.base_width);
+	obs_data_set_int(response, "baseHeight", ovi.base_height);
+	obs_data_set_int(response, "outputWidth", ovi.output_width);
+	obs_data_set_int(response, "outputHeight", ovi.output_height);
+	obs_data_set_double(response, "fps", (double)ovi.fps_num / ovi.fps_den);
+
+	switch (ovi.output_format) {
```

Comment:
Can you put the four `switch` statements in separate functions that return the string value please? Makes them reusable and keeps the main function clutter-free.

Reply:
You can put them in the same file, after include statements & defines and before member functions.

Reply:
Hmm. They currently don't just return a string - the fallback case has them set an integer. If they get refactored into those convenient functions, what should the fallback do? Use a static buffer to represent the integer? Return NULL and let the caller cope?

Reply:
Use the none/unknown enum value as the fallback value. e.g `VIDEO_FORMAT_NONE`

Reply:
Done!